databallpy/load_data/tracking_data/tracab.py

```diff
@@ -0,0 +1,264 @@
+    """
+
+    tracking_data = _get_tracking_data(tracab_loc, verbose)
+    meta_data = _get_meta_data(meta_data_loc)
+
+    return tracking_data, meta_data
+
+
+def _get_tracking_data(tracab_loc: str, verbose: bool) -> pd.DataFrame:
+    """Function that reads tracking data from .dat file and stores it in a pd.DataFrame
+
+    Args:
+        tracab_loc (str): location of the tracking_data.dat file
+        verbose (bool): whether to print info in terminal
+
+    Returns:
+        pd.DataFrame: contains tracking data
+    """
+
+    if verbose:
+        print(f"Reading in {tracab_loc}", end="")
+
+    file = open(tracab_loc, "r")
+    lines = file.readlines()
+    if verbose:
+        print(" - Completed")
+
+    file.close()
+
+    size_lines = len(lines)
+
+    data = {
+        "timestamp": [np.nan] * size_lines,
+        "ball_x": [np.nan] * size_lines,
+        "ball_y": [np.nan] * size_lines,
+        "ball_z": [np.nan] * size_lines,
+        "ball_speed": [np.nan] * size_lines,
+        "ball_status": [None] * size_lines,
+        "ball_posession": [None] * size_lines,
+    }
+
+    if verbose:
+        print("Writing lines to dataframe:")
+
+    for idx, line in enumerate(tqdm(lines)):
+
+        timestamp, players_info, ball_info, _ = line.split(":")
+        data["timestamp"][idx] = int(timestamp)
+
+        players = players_info.split(";")[:-1]
+        for player in players:
+            data = _add_player_data_to_dict(player, data, idx)
+
+        data = _add_ball_data_to_dict(ball_info, data, idx)
+
+    df = pd.DataFrame(data)
+
+    for col in df.columns:
+        if "_x" in col or "_y" in col or "_z" in col or "_speed" in col:
+            df[col] /= 100
+            df[col] = np.round(df[col], 3)
+        else:
+            pass
+
+    df = _insert_missing_rows(df)
+
+    return df
+
+
+def _add_player_data_to_dict(player: str, data: dict, idx: int) -> dict:
+    """Function that adds the data of one player to the data dict for one frame
+
+    Args:
+        player (str): containing data from player
+        data (dict): data dictionary to write results to
+        idx (int): indicates position in data dictionary
+
+    Returns:
+        dict: contains all tracking data
+    """
+
+    team_id, _, shirt_num, x, y, speed = player.split(",")
+
+    team_ids = {0: "home", 1: "away"}
+    team = team_ids.get(int(team_id))
+    if team is None:  # player is unknown or referee
+        return data
+
+    if f"{team}_{shirt_num}_x" not in data.keys():  # create keys for new player
+        data[f"{team}_{shirt_num}_x"] = [np.nan] * len(data["timestamp"])
+        data[f"{team}_{shirt_num}_y"] = [np.nan] * len(data["timestamp"])
+        data[f"{team}_{shirt_num}_speed"] = [np.nan] * len(data["timestamp"])
+
+    data[f"{team}_{shirt_num}_x"][idx] = int(x)
+    data[f"{team}_{shirt_num}_y"][idx] = int(y)
+    data[f"{team}_{shirt_num}_speed"][idx] = float(speed)
+
+    return data
```

Comment:
This function could potentially also be used with other provides, if `team_id, _, shirt_num, x, y, speed = player.split(",")` would be left out. I would suggest adding these to the input of this function, where `speed` for instance could be a optional input since maybe not all providers have this variable.